browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -293,5 +308,294 @@ void BraveTabContainer::PaintChildren(const views::PaintInfo& paint_info) {
+      BraveTabContainer::DropArrow::Position::Horizontal, false);
+
+  // Determine the screen bounds.
+  gfx::Point drop_loc(is_tab_pinned ? center - drop_image->width() / 2 : 0,
+                      is_tab_pinned ? tab->y() - drop_image->height()
+                                    : center - drop_image->height() / 2);
+  ConvertPointToScreen(this, &drop_loc);
+  gfx::Rect drop_bounds(drop_loc.x(), drop_loc.y(), drop_image->width(),
+                        drop_image->height());
+
+  // If the rect doesn't fit on the monitor, push the arrow to the bottom.
+  display::Screen* screen = display::Screen::GetScreen();
+  display::Display display = screen->GetDisplayMatching(drop_bounds);
+  *is_beneath = !display.bounds().Contains(drop_bounds);
+
+  if (*is_beneath) {
+    drop_bounds.Offset(
+        is_tab_pinned ? 0 : drop_bounds.width() + tab->width(),
+        is_tab_pinned ? drop_bounds.height() + tab->height() : 0);
+  }
+
+  return drop_bounds;
+}
+
+gfx::ImageSkia* BraveTabContainer::GetDropArrowImage(
+    BraveTabContainer::DropArrow::Position pos,
+    bool beneath) {
+  using Position = BraveTabContainer::DropArrow::Position;
+  using RotationAmount = SkBitmapOperations::RotationAmount;
+  static base::NoDestructor<
+      base::flat_map<std::pair<Position, bool>, gfx::ImageSkia>>
+      drop_images([] {
+        gfx::ImageSkia* top_arrow_image =
+            ui::ResourceBundle::GetSharedInstance().GetImageSkiaNamed(
+                IDR_TAB_DROP_UP);
+
+        base::flat_map<std::pair<Position, bool>, gfx::ImageSkia>
+            position_to_images;
+
+        position_to_images.emplace(std::make_pair(Position::Vertical, true),
+                                   *top_arrow_image);
+        position_to_images.emplace(
+            std::make_pair(Position::Horizontal, false),
+            gfx::ImageSkiaOperations::CreateRotatedImage(
+                *top_arrow_image, RotationAmount::ROTATION_90_CW));
+        position_to_images.emplace(
+            std::make_pair(Position::Vertical, false),
+            gfx::ImageSkiaOperations::CreateRotatedImage(
+                *top_arrow_image, RotationAmount::ROTATION_180_CW));
+        position_to_images.emplace(
+            std::make_pair(Position::Horizontal, true),
+            gfx::ImageSkiaOperations::CreateRotatedImage(
+                *top_arrow_image, RotationAmount::ROTATION_270_CW));
+        return position_to_images;
+      }());
+  return &drop_images->find(std::make_pair(pos, beneath))->second;
+}
+
+void BraveTabContainer::SetDropArrow(
+    const absl::optional<BrowserRootView::DropIndex>& index) {
+  if (!index) {
+    controller_->OnDropIndexUpdate(absl::nullopt, false);
+    drop_arrow_.reset();
+    return;
+  }
+
+  // Let the controller know of the index update.
+  controller_->OnDropIndexUpdate(index->value, index->drop_before);
+
+  if (drop_arrow_ && (index == drop_arrow_->index())) {
+    return;
+  }
+
+  bool is_beneath;
```

Comment:
need to initialize as `GetDropBounds()` doesn't set the value always.

Reply:
Nice catch1